sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -531,12 +531,31 @@ object GpuToTimestamp extends Arm {
     FIX_SINGLE_DIGIT_SECOND
   )
 
-  def daysScalarSeconds(name: String): Scalar = {
-    Scalar.timestampFromLong(DType.TIMESTAMP_SECONDS, DateUtils.specialDatesSeconds(name))
+  def daysScalarDays(name: String): Scalar = ShimLoader.getSparkVersion match {
+    // In Spark 3.2, special datetime values such as `epoch`, `today`, `yesterday`, `tomorrow`,
+    // and `now` are supported in typed literals only
+    case version if version >= "3.2" =>
```

Comment:
It would be better to add shims for this.

Reply:
If we do keep the version check here (which I am ok with personally because it reduces complexity in the shim layers) then we should use more robust logic for the version check. We have some code in the test suite that could be moved elsewhere.

https://github.com/NVIDIA/spark-rapids/blob/branch-21.10/tests/src/test/scala/com/nvidia/spark/rapids/SparkQueryCompareTestSuite.scala#L1848-L1849

Reply:
+1 to keep the version check.   I would suggest to move @andygrove's  implementation to [ShimVerion ](https://github.com/NVIDIA/spark-rapids/blob/9742ea7790b8f56cac2dba80c1900e72ce76e5a1/sql-plugin/src/main/scala/com/nvidia/spark/rapids/SparkShims.scala#L64)  to make shimversions comparable

Reply:
The reason to use a shim over a version check is mostly databricks and other vendors that pull back "fixes" not directly related to version numbers. I doubt it will happen in this case, but similar things have happened in the past.

Reply:
I think it's something we will find out via the test being broken on databricks. If we go with the ShimVersion comparison implementation, the fix is either an additional condition on databricks shim version compare or it's more involved at which point we will have add a new shim method.

Reply:
I replaced the version dispatch with a shim method.